### DISCUSSION TITLE
The `transformHtml` modification to the `sideBar` code is not taking effect.

### DISCUSSION THREAD

**Discussion by Justin3go** [vuejs/vitepress#4032] [Q&A] [answered]
### Background

I want to implement single-line ellipsis display in the sidebar, and show the title when the mouse hovers over it, like this:

![image](https://github.com/vuejs/vitepress/assets/63507251/a3f3ec09-8eb4-4093-a43f-3ca0c8ec02dc)

The current solution I have in mind is to use the `transformHtml` function to modify the generated HTML code, adding the appropriate title to the corresponding sidebar.

I accomplished this task through simple regex replacement.

![image](https://github.com/vuejs/vitepress/assets/63507251/10d02027-f67c-4255-ae61-d69df405d4ca)

### Question

It is evident that the build artifacts in the `dist` directory have been modified, but when I run `pnpm docs:preview`, the corresponding `title` attribute is not present.

![image](https://github.com/vuejs/vitepress/assets/63507251/9205795d-83cb-4f03-9c78-bd480a199b38)

I want to know why the build artifacts are correct, but the output is not as expected when running. Is there any difference between the two?

### Relevant links

- [Online DEMO](https://feat-sidebar.fav0.pages.dev/)
- [Source Code](https://github.com/Justin3go/FAV0/commit/4f8538a1ab67bf2c4890ec27e8f845e7feaa69e8)

**Comment by brc-dd** [comment, marked as answer]
Patch the component. Using transformHtml to tweak things inside the app element will break hydration.

**Reply by Justin3go** [reply]
Thank you for your response.
